### ISSUE TITLE
[Feature Request] Window scrollbar in terminal vim

### ISSUE THREAD

**Issue by Asheq** [vim/vim#3547]
I think that a scrollbar in each window would be a big UX win.

I'm imagining the scrollbar as a single column, on the very right hand-side of each window, as is standard.

This could be very similar to the scrollbar in the insert completion popup menu. I don't care whether the "gripper" on the scrollbar is interactive or not (e.g. whether you can use the mouse to grab and move the scrollbar). The scrollbar could simply act as visual feedback, and I would be very happy.

**Comment by chrisbra** [comment]
What aren’t you missing from the default scrollbar? Check the guioptions setting

**Comment by Asheq** [comment]
Doesn't `guioptions` only apply to GUI Vim (e.g. MacVim)? I would like see scrollbars in terminal Vim. Is this already possible?

**Comment by chrisbra** [comment]
Yes that was not entirely obvious from your question. You can use the sign feature for that, but it will take away two columns. 

See https://github.com/chrisbra/DynamicSigns
Or https://superuser.com/a/445233

For an example.

I don’t think this is a valid issue though.

**Comment by tyru** [comment]
Sorry, how do I implement scrollbar by using your plugin?
I see the help of your plugin, but I don't know how.

And sign can add columns to only left,
and it may conflict with other sign plugin.
So I also want scrollbar in CUI Vim like the one which popup completion menu has.

**Comment by k-takata** [comment]
I think that supporting scrollbars on terminal Vim is a reasonable and valid feature request.
I have seen this kind of request several times.
I don't think it is easy to implement, though.

**Comment by chrisbra** [comment]
okay convinced, let's reopen

**Comment by chdiza** [comment]
I don't see the point of this.  `&ruler` (or `&statusline`) already puts a visual aid in the bottom right about how far through a file one is at the moment.

I'm sure a custom function in `&statusline` could even change the highlighting color for that percentage indicator depending on its value.  This would make it stand out a bit more.

**Comment by chrisbra** [comment]
There are some similar plugins mentioned in the corresponding Neovim issue https://github.com/neovim/neovim/issues/9132#issuecomment-430530887
